src/atlas/functionspace/PointCloud.cc

```diff
@@ -58,18 +65,13 @@ PointCloud::PointCloud(const std::vector<PointXYZ>& points) {
 
 PointCloud::PointCloud(const Field& lonlat): lonlat_(lonlat) {}
 
-PointCloud::PointCloud(const Field& lonlat, const Field& ghost): lonlat_(lonlat), ghost_(ghost) {}
+PointCloud::PointCloud(const Field& lonlat, const Field& ghost): lonlat_(lonlat), ghost_(ghost) {
+    setupHaloExchange();
+}
 
 PointCloud::PointCloud(const FieldSet & flds): lonlat_(flds["lonlat"]),
-  ghost_(flds["ghost"]), remote_index_(flds["remote_index"]), partition_(flds["partition"])
-{
-    ATLAS_ASSERT(ghost_.size() == remote_index_.size());
-    ATLAS_ASSERT(ghost_.size() == partition_.size());
-    halo_exchange_.reset(new parallel::HaloExchange());
-    halo_exchange_->setup(array::make_view<int, 1>( partition_).data(),
-                          array::make_view<idx_t, 1>(remote_index_).data(),
-                          REMOTE_IDX_BASE,
-                          ghost_.size());
+  ghost_(flds["ghost"]), remote_index_(flds["remote_index"]), partition_(flds["partition"]) {
```

Comment:
I personally would have kept the old functionality for this constructor to allow for special cases.

Reply:
@MarekWlasak  please note that the constructor `PointCloud(const FieldSet & flds)` that uses four fields for the intialization has not been modified -- see [lines 422-429](https://github.com/JCSDA-internal/atlas/blob/feature/pointcloud_mpi_01/src/atlas/functionspace/PointCloud.cc#L422).
The introduction of the function `setupHaloExchange()` and the check [here](https://github.com/JCSDA-internal/atlas/blob/feature/pointcloud_mpi_01/src/atlas/functionspace/PointCloud.cc#L309) allow to avoid redundant code by sharing the procedure to set up the halo exchange across constructors with different signature.